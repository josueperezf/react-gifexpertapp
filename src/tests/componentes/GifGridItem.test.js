import { shallow } from "enzyme";
import GifGridItem from "../../componentes/GifGridItem";

describe('prueb para el componente <GifGridItem/>',()=>{
    const img = {
        "id": "oxbNORcXx76F2",
        "title": "opm GIF",
        "url": "https://media0.giphy.com/media/oxbNORcXx76F2/giphy.gif?cid=ed0298b533ss8ce6vzekx98hbb4mlh4k8pvhahiumjlgdlje&rid=giphy.gif"
    };
    const wrapper = shallow(<GifGridItem  key={img.id}  {...img} />);

    test('debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de tener un parrafo con el title', () => {
        const etiquetaP = wrapper.find('p').text().trim();
        expect(img.title).toBe(etiquetaP);
    });
    
    test('la imagen del componente debe tener la url y el alt que el envio', () => {
        const etiquetaImg = wrapper.find('img');
        // props sirve para obtener TODAS las propiedades de una etiqueta html
        // prop sirve para obtener UNA propiedad de una etiqueta html
        const altImg = etiquetaImg.prop('alt');
        const srcImg = etiquetaImg.prop('src');
        expect(img.title).toBe(altImg);
        expect(img.url).toBe(srcImg);
    });
    
    test('el div debe contener la clase animate__fadeIn', () => {
        const claseCssBuscar    = 'animate__fadeIn';
        const clasesEnDiv       = wrapper.find('div').prop('className');
        expect( clasesEnDiv.includes(claseCssBuscar)).toBe(true);
    });
    
});
