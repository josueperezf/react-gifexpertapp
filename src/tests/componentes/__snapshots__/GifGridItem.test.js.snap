// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueb para el componente <GifGridItem/> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="opm GIF"
    src="https://media0.giphy.com/media/oxbNORcXx76F2/giphy.gif?cid=ed0298b533ss8ce6vzekx98hbb4mlh4k8pvhahiumjlgdlje&rid=giphy.gif"
  />
  <p>
    opm GIF
  </p>
</div>
`;
